common/rpc/interceptor/caller_info.go

```diff
@@ -35,32 +35,48 @@ import (
 )
 
 type (
-	CallerInfoInterceptor struct {
+	CallerInfo struct {
 		namespaceRegistry namespace.Registry
 	}
 )
 
-var _ grpc.UnaryServerInterceptor = (*CallerInfoInterceptor)(nil).Intercept
+var _ grpc.UnaryServerInterceptor = (*CallerInfo)(nil).Intercept
 
-func NewCallerInfoInterceptor(
+func NewCallerInfo(
 	namespaceRegistry namespace.Registry,
-) *CallerInfoInterceptor {
-	return &CallerInfoInterceptor{
+) *CallerInfo {
+	return &CallerInfo{
 		namespaceRegistry: namespaceRegistry,
 	}
 }
 
-func (i *CallerInfoInterceptor) Intercept(
+func (i *CallerInfo) Intercept(
 	ctx context.Context,
 	req interface{},
 	info *grpc.UnaryServerInfo,
 	handler grpc.UnaryHandler,
 ) (interface{}, error) {
+	ctx = PopulateCallerInfo(
+		ctx,
+		func() string { return string(MustGetNamespaceName(i.namespaceRegistry, req)) },
+		func() string { return api.MethodName(info.FullMethod) },
+	)
+
+	return handler(ctx, req)
+}
+
+// PopulateCallerInfo gets current caller info value from the context and updates any that are missing.
+// Namespace name and method are passed as functions to avoid expensive lookups if those values are already set.
+func PopulateCallerInfo(
```

Comment:
Looking at what this does, I don't think we need it for Nexus requests. Is this for propagating gRPC metadata to matching and history?

Reply:
It is used to propagate values from frontend to other services, so I think we should include it to avoid breaking any assumptions.

Reply:
Works for me.